File: frontend/src/types/agent.ts
export interface Agent {
  id: number;
  student_id: number;
  name: string;
  avatar_url: string;
  description: string;
  created_at: string;
  updated_at: string;
}

// 添加一个工具函数来处理 URL
export const getFullImageUrl = (url: string | null) => {
  if (!url) return null;
  if (url.startsWith('http')) return url;
  return `${process.env.NEXT_PUBLIC_API_BASE_URL}${url}`;
}; 

File: backend/app/schemas/agent.py
from pydantic import BaseModel
from typing import List, Optional
from datetime import datetime

class AgentBase(BaseModel):
    student_id: int
    name: str
    avatar_url: Optional[str] = None
    description: Optional[str] = None

class AgentCreate(AgentBase):
    pass

class AgentUpdate(AgentBase):
    id: int

class AgentInDB(AgentBase):
    id: int
    created_at: datetime
    updated_at: datetime

    class Config:
        from_attributes = True